x/distribution/types/dec_coin.go

```diff
@@ -0,0 +1,141 @@
+package types
+
+import (
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// Coins which can have additional decimal points
+type DecCoin struct {
+	Denom  string  `json:"denom"`
+	Amount sdk.Dec `json:"amount"`
+}
+
+func NewDecCoin(coin sdk.Coin) DecCoin {
+	return DecCoin{
+		Denom:  coin.Denom,
+		Amount: sdk.NewDecFromInt(coin.Amount),
+	}
+}
+
+// Adds amounts of two coins with same denom
+func (coin DecCoin) Plus(coinB DecCoin) DecCoin {
+	if !(coin.Denom == coinB.Denom) {
```

Comment:
`coin.Denom != coinB.Denom` ?

Reply:
Also, maybe we should panic instead of returning the first coin. Is there ever a case where we'd want that behavior?

Reply:
yeah I just adapted this code from the Coins type.. but I agree we should panic, I'll spiff dis up